core/web/resolver/node_test.go

```diff
@@ -0,0 +1,271 @@
+package resolver
+
+import (
+	"database/sql"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/guregu/null.v4"
+
+	"github.com/smartcontractkit/chainlink/core/chains/evm"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/core/utils"
+)
+
+func Test_NodeQuery(t *testing.T) {
+	t.Parallel()
+
+	query := `
+		query GetNode {
+			node(id: "200") {
+				... on Node {
+					name
+					wsURL
+					httpURL
+				}
+				... on NotFoundError {
+					message
+					code
+				}
+			}
+		}`
+	notFoundQuery := `
+		query GetNode {
+			node(id: "1") {
+				... on Node {
+					name
+					wsURL
+					httpURL
+				}
+				... on NotFoundError {
+					message
+					code
+				}
+			}
+		}`
```

Comment:
Small nit: I don't think you need a separate query for this. In your second test case you can just pass in the same query, but just mock an error response like you already do: `f.Mocks.evmORM.On("Node", int32(1)).Return(types.Node{}, sql.ErrNoRows)` and it should go through the NotFoundError code path.